enhancements/insights/pulling-data-from-ocm.md

```diff
@@ -0,0 +1,96 @@
+---
+title: pulling-and-exposing-data-from-ocm
+authors:
+  - "@tremes"
+reviewers:
+  - "@sbose78"
+  - "@0sewa0"
+  - "@inecas"
+  - "@smarterclayton"
+approvers:
+  - "@sbose78"
+  - "@smarterclayton"
+creation-date: 2021-03-04
+last-updated: 2021-03-09
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# Insights Operator pulling and exposing data from the OCM API
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement will enable the Insights Operator to pull the data (SCA certs)
```

Comment:
What "data" is being pulled?

Reply:
SCA (Simple Content Access) certificates from RHSM. These certs are required by OpenShift clusters to pull RHEL content for builds (from https://issues.redhat.com/browse/SDB-2197)

Reply:
How will those certificates make it into a cluster in an environment where OCM is not present?

Does the insights operator use the certificates itself?

Reply:
They won't make it into a cluster. No the Insights operator doesn't use or require them.

Reply:
> How will those certificates make it into a cluster in an environment where OCM is not present?

I assume that these clusters would have a Satellite instance where the SCA can be obtained.
For a first pass I think this use case can be considered out of scope - admins can manually obtain the SCA from satellite, add it as a secret in the `openshift-config`, and then create the `Share` object to make it accessible in the cluster.

Reply:
OK, I was asking about how non-OCM clusters do the same thing because it didn't make sense to me for the insights operator to do it just in 1 case. I think the explanation about why the insights operator needs to do it from another thread resolves my question about that. I agree that we can leave other use cases to future enhancements.

I am curious, though, about why we don't make use of OCM features like policy or hive features like syncsets to push certificates down into the cluster instead of making a special case for the operator to pull them. Was that approach explored at all?

Reply:
We did not expore OCM or hive. Two thoughts here regarding using OCM or Hive:

- OpenShift Cluster Management (OCM) is an add on that not every OpenShift cluster has. Not as sure about hive other than it is not to my knowledge used by all customers/clusters
- SCA certs change regularly (basically every time the customer makes changes to their subscriptions). Something needs to fetch the certificate and update it automatically - we arrived at the insights operator being the best tool we had lying around.

Reply:
OK, I was conflating OCM and ACM. Now that I understand they're different things, I see why we need something inside the cluster pulling the certs down.

Reply:
@dhellmann are you ok resolving this convo?